### ISSUE TITLE
T10.26. Чи може клас зовсім не мати деструкторів?

### ISSUE THREAD

**Issue by WAG13** [kzhereb/knu-is-ooop2019#26]

**Comment by i1red** [comment]
Ні, якщо ви не описали жодного деструктора, то буде викликатися деструктор, створений компілятором за замовчуванням

**Comment by solovdaria** [comment]
Ні, не може. Якщо в класі не буде явно прописаного деструктора, то компілятор створить деструктор за замовчуванням, визначений неявним способом.

**Comment by AntonHladkiy** [comment]
Ні, якщо деструктора немає буде викликаний деструктор за замовчуванням

**Comment by RostyslavMV** [comment]
Тут аналогічна ситуація як з конструкторами, сучасні компілятори створять його, проте краще уникати цього, щоб уникнути проблем з непередбачуваною поведінкою деструктора, створеного компілятором.

**Comment by ilyakorenevskyi** [comment]
Як і з конструктором, якщо ви не написали його самі, його створить компілятор, але краще писати самому, щоб все працювало як треба

**Comment by WAG13** [comment]
За замовчуванням клас уже має деструктор, тому прописувати його не обов'язково.
